ScoreSaber/UI/Leaderboard/ScoreSaberLeaderboardViewController.cs

```diff
@@ -238,105 +224,318 @@ private void uploadDaemon_UploadStatusChanged(UploadStatus status, string status
+            new IconSegmentedControl.DataItem(Utilities.FindSpriteInAssembly("ScoreSaber.Resources.country.png"), "Area")
+        };
+#pragma warning restore CS0618 // Type or member is obsolete
+            }
+        }
+
+        private void ShowRichPresenceDisclaimer() {
+            var info = new GenericYesOrNoModal.YesOrNoModalInfo("Rich Presence", delegate () {
+                Plugin.Settings.enableRichPresence = true;
+                Plugin.Settings.hasAcceptedRichPresenceDisclaimer = true;
+                _scoresaberRichPresence.Initialize();
+                _parserParams.EmitEvent("close-modals");
+            }, delegate () {
+                Plugin.Settings.enableRichPresence = false;
+                Plugin.Settings.hasAcceptedRichPresenceDisclaimer = true;
+                _parserParams.EmitEvent("close-modals");
+
+            }, "Rich Presence is a feature that allows you to show your current status in Beat Saber to your friends. Would you like to enable this?\n(You can turn it off anytime in the ScoreSaber settings menu)");
+            _genericYesOrNoModal.Show(info);
+            _parserParams.EmitEvent("present-yes-no-modal");
+        }
+
+        protected override void DidActivate(bool firstActivation, bool addedToHierarchy, bool screenSystemEnabling) { 
+            base.DidActivate(firstActivation, addedToHierarchy, screenSystemEnabling);
+            if (!base.isActiveAndEnabled) return;
+            if (!_platformLeaderboardViewController) return;
+            if (firstActivation) {
+                _panelView.statusWasSelected = delegate () {
+                    if (_leaderboardService.currentLoadedLeaderboard == null) { return; }
+                    _parserParams.EmitEvent("close-modals");
+                    Application.OpenURL($"https://scoresaber.com/leaderboard/{_leaderboardService.currentLoadedLeaderboard.leaderboardInfoMap.leaderboardInfo.id}");
+                };
+
+                _panelView.rankingWasSelected = delegate () {
+                    _parserParams.EmitEvent("close-modals");
+                    _parserParams.EmitEvent("show-profile");
+                    _profileDetailView.ShowProfile(_playerService.localPlayerInfo.playerId).RunTask();
+                };
+
+                _container.Inject(_profileDetailView);
+                _ImageHolders.ForEach(holder => holder.ClearSprite());
+                activated = true;
+                OnIconSelected(null, 0);
+                _promoBanner.CreatePromoBanner();
+                _promoBanner._floatingScreen.gameObject.SetActive(true);
+                _promoBanner._promoBannerView.GetComponent<CanvasGroup>().alpha = 1;
+            }
+
+            UnityMainThreadTaskScheduler.Factory.StartNew(async () => {
+                if (!Plugin.Settings.hasAcceptedRichPresenceDisclaimer) {
+                    await Task.Delay(500);
```

Comment:
Do we have a runtime issue if this delay doesn't exist or is it purely for UX? If it's for UX, this is fine. But if we're avoiding a runtime issue this isn't a stable fix and should be addressed correctly.

Reply:
without the delay, the modal doesnt have the foreground effect for some reason. Im not sure why

Reply:
All it needs is a single millisecond after some testing

Reply:
This implies that we're needing to wait a frame. Try `await Task.Yield();` instead.

Or, use a coroutine which begins with `yield return null;` to wait until the next frame before `ShowRichPresenceDisclaimer()`